tests/functional/test_basic.py

```diff
@@ -195,7 +189,9 @@ def test_python_makefile(pyi_builder):
 
 
 def test_set_icon(pyi_builder, data_dir):
-    icon_dir = os.path.join(data_dir, 'icons')
+    # Note that the name test_set_icon (not just test_icon) for prevents
+    # data_dir from copying data/icon to the tmpdir.
+    icon_dir = os.path.join(data_dir.strpath, 'icons')
```

Comment:
I see things a bit differently. My intention is to have test code operate on `data/` directly; code under test should operate only on `testdir/data`. Separating these two, at least to me, keeps the code a bit cleaner, while combining them (which is your suggestion, as I understand it) would blur this distinction. However, as you note, preventing access to `data/` by providing a path only to `tmpdir/data` does simplify the code in another sense: there's only one path for `data_dir` to return.

Your thoughts?